### PR TITLE
build(docker): Use redis-cluster for compose, consolidate Travis

### PR DESCRIPTION
**Pull request by BYK**
Enables redis-cluster for the Docker Compose test setup on GCB and removes the custom logic from TravisCI to use the same setup.

This is a follow up to #727. A logical next step would be to stop building a docker image inside Travis and trigger the Travis builds from GCB once the image is built. That said a better option would be to simply moving everything to GCB for efficiency.

### CODE REVIEW

File: .travis.yml
Comment (tkaemming) [inline .travis.yml]: It took me a second to realize what this was doing, but this is pretty clever.

File: .travis.yml
Comment (tkaemming) [inline .travis.yml]: I assume that `docker-compose` is doing the variable substitution here — can `SNUBA_IMAGE` have a default?

**Review by tkaemming — APPROVED** [review]
This is good, thanks for following through with this.

File: .travis.yml
Comment (BYK) [reply .travis.yml]: We can even cheat further and do
```yaml
- docker pull us.gcr.io/sentryio/snuba:$TRAVIS_COMMIT || docker pull getsentry/snuba:latest || true
- docker build -t getsentry/snuba . --cache-from us.gcr.io/sentryio/snuba:$TRAVIS_COMMIT --cache-from getsentry/snuba:latest
```

If GCB finishes earlier than Travis (quite unlikely) we'd just pull from that image.

File: .travis.yml
Comment (BYK) [reply .travis.yml]: Unfortunately, having default values in `docker-compose.yaml` files for environment variables was removed in v3, so no ☹️

**Comment by BYK** [comment]
@JTCunning will wait for your green light before merging. After that I'm hoping to end this dual `cloudbuild.yaml` thing.

**Review by JTCunning — APPROVED** [review]
🍪

**Pull request merged**
